DiscordChatExporter.Core/Exporting/Filtering/BooleanMessageFilter.cs

```diff
@@ -0,0 +1,27 @@
+﻿using DiscordChatExporter.Core.Discord.Data;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public class BooleanMessageFilter : MessageFilter
```

Comment:
In parsing/formal language terminology, this type of syntax node is usually called a binary expression (i.e. 'equals' binary expression, 'and' binary expression, 'greater than' binary expression, etc). Maybe we can call it something like `BinaryExpressionMessageFilter`?